test/client_ui_test.py

```diff
@@ -94,21 +113,57 @@ def test_close():
     pb.size = 1 * GiB
     pb.update(512 * MiB)
     pb.close()
+    f.last = None
 
     # Once closed, update does not redraw.
-    f.last = None
     pb.update(512 * MiB)
     assert f.last is None
 
+    # Changing size does nothing.
+    pb.size = 2 * GiB
+    assert f.last is None
+    assert pb.size == 1 * GiB
+
+    # Changing phase does nothing.
+    pb.phase = "new phase"
+    assert f.last is None
+    assert pb.phase is None
+
     # Closing twice does not redraw.
     pb.close()
     assert f.last is None
 
 
 def test_contextmanager():
     f = FakeFile()
-    with client.ProgressBar(GiB, output=f) as pb:
+    with client.ProgressBar(size=GiB, output=f) as pb:
         pb.update(GiB)
         assert f.last.endswith("\r")
 
     assert f.last.endswith("\n")
+
+
+def test_error_phase_default():
+    f = FakeFile()
+    pb = client.ProgressBar(phase="running command", size=GiB, output=f)
+    with pytest.raises(RuntimeError):
+        with pb:
+            raise RuntimeError
+
+    assert f.last.endswith("\n")
+    assert pb.phase == "command failed"
+
+
+def test_error_phase_custom():
+    f = FakeFile()
+    pb = client.ProgressBar(
+        phase="starting operation",
+        error_phase="operation failed",
+        size=GiB,
+        output=f)
+    with pytest.raises(RuntimeError):
+        with pb:
+            raise RuntimeError
+
+    assert f.last.endswith("\n")
+    assert pb.phase == "operation failed"
```

Comment:
When reading the code I was wondering, if/how this error-phase actually gets drawn. I saw it then, but maybe the test should verify that too, so it's not just some internal state.

Reply:
Fixed in current version, we test also the output.